st2auth/st2auth/backends/__init__.py

```diff
@@ -38,16 +46,12 @@ def get_backend_instance(name):
     :param name: Backend name.
     :type name: ``str``
     """
-    if name not in VALID_BACKEND_NAMES:
-        raise ValueError('Invalid authentication backend specified: %s', name)
-
-    module = importlib.import_module(BACKEND_MODULES[name])
-    classes = _get_classes_in_module(module=module)
-
     try:
-        cls = [klass for klass in classes if klass.__name__.endswith('AuthenticationBackend')][0]
-    except IndexError:
-        raise ValueError('"%s" backend module doesn\'t export a compatible class' % (name))
+        manager = DriverManager(namespace=BACKENDS_NAMESPACE, name=name,
+                                invoke_on_load=False)
+    except RuntimeError as e:
+        raise ValueError('Invalid authentication backend specified: %s (%s)' %
```

Comment:
I did include it in the new exception we throw (`str(e)`), but I can also log the original exception using. LOG.exception, but this will result in exception being logged twice.